imports/server/gdriveChangeFetcher.ts

```diff
@@ -0,0 +1,151 @@
+/* eslint-disable camelcase */
+import { Meteor } from 'meteor/meteor';
+import type { drive_v3 } from '@googleapis/drive';
+import Flags from '../Flags';
+import DocumentActivities from '../lib/models/DocumentActivities';
+import Documents from '../lib/models/Documents';
+import FeatureFlags from '../lib/models/FeatureFlags';
+import DriveClient from './gdriveClientRefresher';
+import ignoringDuplicateKeyErrors from './ignoringDuplicateKeyErrors';
+import DriveChangesPageTokens from './models/DriveChangesPageTokens';
+import Locks, { PREEMPT_TIMEOUT } from './models/Locks';
+
+const ACTIVITY_GRANULARITY = 5 * 60 * 1000; // milliseconds
+
+async function recordDriveChange(file: Pick<drive_v3.Schema$File, 'id' | 'modifiedTime'>) {
+  if (!file.id || !file.modifiedTime) {
+    return;
+  }
+
+  const document = Documents.findOne({ 'value.id': file.id });
```

Comment:
We might want to add an index on this field if we're querying by it?

Reply:
Huh yeah I guess I assumed we had this already. Fixing.